docs/compatibility.md

```diff
@@ -0,0 +1,77 @@
+# StableHLO Compatibility
+
+StableHLO is a backward compatible ML compute opset inspired by HLO/MHLO.
+This document explains the kind and the extent of the compatibility guarantees
+that StableHLO provides, based on the process established in
+[the compatibility RFC](../rfcs/20220912-compatibility.md).
+
+## Versions
+
+The current version of StableHLO is **0.9.0**.
+
+In the 0.x.x series, StableHLO is providing limited compatibility guarantees
+as described in the Guarantees section. In H2 2023, we are planning to release
+StableHLO 1.0.0, which will provide full compatibility guarantees - see the
+Future work section for more details.
+
+## Guarantees
+
+**1 month of backward compatibility:** VHLO program serialized by an old version
+of libStablehlo has the same semantics* when deserialized by a new version of
+libStablehlo if these versions are built from openxla/stablehlo commits which
+are less than 1 month apart.
+
+**1 month of forward compatibility:** VHLO program serialized by a new version
+of libStablehlo has the same semantics* when deserialized by an old version of
+libStablehlo if these versions are built from openxla/stablehlo commits which
+are less than 1 month apart, unless the program is using new features introduced
+since the old version.
+
+\* VHLO is a dialect that contains versioned snapshots of the StableHLO opset.
+It can be obtained from the StableHLO dialect via
+[--stablehlo-legalize-to-vhlo](../stablehlo/tests/stablehlo_legalize_to_vhlo.mlir),
+and its semantics is defined by [the StableHLO spec](spec.md).
+
```

Comment:
Should we add a section on "How to use VHLO".

## Using the VHLO Dialect

### How to create a VHLO stable artifact

VHLO bytecode is the stable serialization format that offers compatibility guarantees. Stable artifacts can be created and loaded as follows:

```
# Create stable bytecode artifact:
$ stablehlo-opt <file.mlir> --stablehlo-legalize-to-vhlo --emit-bytecode > stable_file.mlir.bc

# Load program (guaranteed within compatibility window):
# Upgrade if needed and convert to StableHLO 
$ stablehlo-opt stable_file.mlir.bc --vhlo-to-version='target=current' --vhlo-legalize-to-stablehlo
```

### Target previous versions of libStablehlo for forward compatibility

If forward compatibility is needed, a previous version of libStablehlo can be targeted. If conversion and serialization succeeds, the resulting artifact can be used on both old and new libStablehlo consumers.

```
# Create stable bytecode artifact for libStablehlo version 0.X.Y:
# This will fail if features not compatible with 0.X.Y are used.
$ stablehlo-opt <file.mlir> --stablehlo-legalize-to-vhlo --vhlo-to-version='target=0.X.Y' --emit-bytecode > stable_file.mlir.bc

# Load program (guaranteed within compatibility window)
# Works on both the old and new versions of libStablehlo
$ stablehlo-opt stable_file.mlir.bc --vhlo-to-version='target=current' --vhlo-legalize-to-stablehlo
```

Reply:
Discussed offline. We'll add vhlo.md that will be linked from compatibility.md.